tests/unit/aiplatform/test_lro.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2020, Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+
+from google.api_core import operation
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import lro
+from google.cloud.aiplatform_v1beta1.services.model_service.client import (
+    ModelServiceClient
+)
+from google.longrunning import operations_pb2
+from google.protobuf import struct_pb2 as struct
+
+TEST_OPERATION_NAME = "test/operation"
+
+
+class AiPlatformResourceNounImpl(base.AiPlatformResourceNoun):
+    client_class = ModelServiceClient
+    _is_client_prediction_client = False
+
+
+def make_operation_proto(
+    name=TEST_OPERATION_NAME, metadata=None, response=None, error=None, **kwargs
+):
+    operation_proto = operations_pb2.Operation(name=name, **kwargs)
+
+    if metadata is not None:
+        operation_proto.metadata.Pack(metadata)
+
+    if response is not None:
+        operation_proto.response.Pack(response)
+
+    if error is not None:
+        operation_proto.error.CopyFrom(error)
+
+    return operation_proto
+
+
+def make_operation_future(client_operations_responses=None):
+    if client_operations_responses is None:
+        client_operations_responses = [make_operation_proto()]
+
+    refresh = mock.Mock(spec=["__call__"], side_effect=client_operations_responses)
+    refresh.responses = client_operations_responses
+    cancel = mock.Mock(spec=["__call__"])
+    operation_future = operation.Operation(
+        client_operations_responses[0],
+        refresh,
+        cancel,
+        result_type=struct.Struct,
+        metadata_type=struct.Struct,
+    )
+
+    return operation_future
+
+
+def test_constructor():
+    operation_future = make_operation_future()
+    test_lro = lro.LRO(operation_future)
+
+    assert test_lro.operation_future.operation.name == TEST_OPERATION_NAME
+    assert test_lro.operation_future._result_type is struct.Struct
+    assert test_lro.operation_future.done() is False
+    assert len(test_lro.operation_future._done_callbacks) is 0
+
+
+def test_constructor_with_update():
+    operation_future = make_operation_future()
+    resource_noun_obj = AiPlatformResourceNounImpl()
+    result_key = "name"
+    api_get = mock.Mock(spec=["__call__"])
+    test_lro = lro.LRO(operation_future, resource_noun_obj, result_key, api_get)
+
+    assert len(test_lro.operation_future._done_callbacks) is 1
```

Comment:
Need a test that verifies that the callback is actually called and that `resource_noun_obj._gca_resource` is correctly populated.

Reply:
we can verify that the callback is actually called but wouldn't it make more sense as unit test of ga_operation.Operation or google.api_core.future.polling.PollingFuture?

Reply:
This `LRO` class is depending on how ga_operation handles callbacks, for example we are assuming that it would be called with the correct context (with `resource_noun_obj` in scope, for example).  This depends on how ga_operation implements callback, and we should make sure it works in the way we need it to.  For instance, python concurrent.future's callback includes a `context` parameter, implying that some variables would not be in scope.